site-src/geps/gep-1762.md

```diff
@@ -0,0 +1,309 @@
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1beta1
+kind: Gateway
+metadata:
+  name: my-gateway
+spec:
+  addresses:
+  - type: IPAddress
+    value: 1.1.1.1
+  gatewayClassName: example
+  listeners:
+  - name: default
+    port: 80
+    protocol: HTTP
+```
+
+This would generate a `Service` with `clusterIP` or `loadBalancerIP`, depending on the Service type.
+
+Note: this differs from the selection of "manual deployment", which doesn't use `IPAddress` type.
+
+#### Labels and Annotations
+
+Labels and annotations SHOULD be copied from the `Gateway` to generated resources.
+
+Note this may mean an annotation intended for a `Service` may end up on a `Deployment` (for example).
+This is typically not a concern.
+However, if an implementation is aware of specific meanings of certain labels or annotations, they MAY
+exclude these from irrelevant resources.
+
+#### Arbitrary Customization
+
+Currently, to provide arbitrary customizations, Gateway API provides a few mechanisms:
+
+* GatewayClass has a `parametersRef` field, allowing to point to an arbitrary other resource.
+
+  This *can* be used to make arbitrary customizations, but comes with a number of issues:
+  * GatewayClass is cluster scoped. This means that any user configuring a namespace-scoped Gateway
+    may need to make a new cluster-scoped resources, which makes RBAC extremely challenging.
+    An analogy would be to have a `DeploymentClass` to configure CPU requests, instead of a field in each `Deployment`.
+  * Many levels of indirection. To configure my `Gateway`, I need a `Gateway`, `GatewayClass`, and `SomeCustomParamsType`.
+  * This is intended to be immutable according to the `GatewayClass` spec. This makes it challenging to operate and implement.
+  * There is no common type to reference; each implementation would make their own.
+
+* Annotations are always an option, but one that is counter to many Gateway API goals.
+
+* Policy attachment could be used to modify a `Gateway`, but this is a fairly unintended usage of policy attachment.
+
+* Modifying and merging user values and controller values.
+  Users can manually create the generated resources and have them merged in with the controller values.
+
+  With careful use of Server Side Apply, this could be done in any order. However, this comes with downsides:
+  * No good way to have implementation provide "defaults" and "overrides".
+  * Order may matter; if implementation is first we may temporarily have unsafe settings (for example, run as root later overwritten to run as non-root).
+  * This type of ordering dependency is tricky operationally, especially with GitOps.
+
+None of these mechanisms provide a satisfying solution.
+
+Some options:
+
+* Add a `infrastructure` field into `Gateway`. This would have a few common configuration settings in the top level, then delegate to an arbitrary type:
+
+  ```yaml
+  spec:
+    infrastructure:
+      # Define the scope of the Gateway (GEP-1651)
+      scope: InCluster
```

Comment:
I think I'd personally rather see this as per-address config or a GatewayClass field.